src/modules/addNewChain/AddNewChain.tsx

```diff
@@ -0,0 +1,173 @@
+// React and other libraries
+import { FC, useState } from 'react';
+
+import { useSelector } from 'react-redux';
+import { useFormik } from 'formik';
+import * as yup from 'yup';
+import { css } from 'styled-components';
+import { MdError } from 'react-icons/md';
+
+import useToast from 'hooks/useToast';
+import { useInitiateNewChain } from 'queries';
+
+import { NewAddress } from './components/NewAddress';
+import { ChangeNetwork } from './components/ChangeNetwork';
+import { VerifyAliasChain } from './components/VerifyAliasChain';
+import UnlockProfileWrapper, { UNLOCK_PROFILE_TYPE } from 'components/chat/unlockProfile/UnlockProfileWrapper';
+import { Box, Text } from 'blocks';
+import { Stepper } from 'common';
+
+import { UserStoreType } from 'types';
+import { ActiveStepKey, NewChainAddressValue } from './AddNewChain.types';
+
+import { isValidAddress } from 'helpers/ValidationHelper';
+import { convertAddressToAddrCaip } from 'helpers/CaipHelper';
+import { addNewChainSteps } from './AddNewChain.constants';
+
+const AddNewChain: FC = () => {
+  const [activeStepKey, setActiveStepKey] = useState<ActiveStepKey>('newaddress');
+  const [completedSteps, setCompletedSteps] = useState<Array<ActiveStepKey>>(['newaddress']);
+
+  const toast = useToast();
+  const { mutate: initiateNewChain, isPending, isError } = useInitiateNewChain();
+  const { userPushSDKInstance } = useSelector((state: UserStoreType) => state.user);
+
+  const validationSchema = yup.object().shape({
+    alias: yup
+      .string()
+      .required('Address is required')
+      .test('is-valid-address', 'Invalid wallet address', isValidAddress),
+    chainId: yup.string().required('ChainId is required'),
+  });
+  const formik = useFormik<NewChainAddressValue>({
+    initialValues: {
+      alias: '',
+      chainId: '11155111',
+    },
+    validationSchema: validationSchema,
+    onSubmit: (values) => {
+      handleInitiate(values.alias, values.chainId);
+    },
+  });
```

Comment:
We decided the validation schema, form types and initial values goes to the AddNewChain.form.ts

Reply:
@rohitmalhotra1420  but useFormik can only be used in a react component.

Reply:
I am not talking out moving out the hook but  validation schema, form types and initial values from the component to another file